src/QuadVelocityController.cpp

```diff
@@ -142,36 +160,53 @@ bool QuadVelocityController::update(const ros::Time& time,
     }
 
     // Print the velocity and throttle information
-    ROS_INFO("Vz: %f Vx: %f Vy: %f Vyaw: %f",
-             velocity.linear.z,
-             velocity.linear.x,
-             velocity.linear.y,
-             velocity.angular.z);
+    ROS_INFO("Vz: %f Vx: %f Vy: %f",
+             velocity.z,
+             velocity.x,
+             velocity.y);
     ROS_INFO("Throttle: %f Pitch: %f Roll: %f Yaw: %f",
              uav_command.throttle,
              uav_command.data.pitch,
              uav_command.data.roll,
              uav_command.data.yaw);
 
+    last_update_time_ = time;
     return true;
 }
 
 bool QuadVelocityController::waitUntilReady()
 {
-    geometry_msgs::TransformStamped transform;
-    bool success = getTransformAtTime(
-                       transform,
-                       ros::Time(0),
-                       ros::Duration(INITIAL_TRANSFORM_WAIT_SECONDS));
+    const ros::Time start_time = ros::Time::now();
+    while (ros::ok()
+           && odometry_msg_queue_.empty()
+           && ros::Time::now() < start_time + max_initial_wait_) {
+        ros::spinOnce();
+        ros::Duration(0.001).sleep();
+    }
 
-    last_transform_stamped_ = transform;
+    if (odometry_msg_queue_.empty()) {
+        ROS_ERROR("Failed to fetch initial velocity");
+        return false;
+    }
+
+    geometry_msgs::TransformStamped transform;
+    bool success = getTransformAtTime(transform,
+                                      ros::Time(0),
+                                      max_initial_wait_);
 
     if (!success)
     {
         ROS_ERROR("Failed to fetch initial transform");
+        return false;
+    }
+    else
+    {
+        // Mark the last update time as 1ns later than the odometry message,
+        // because we should always have a message older than the last update
```

Comment:
Because it's not actually so we wait for a second orientation, it's because the invariant on the odometry buffer is that there's always a message older than the last update time.  I'll put a comment explaining that by the declaration.